yagb-core/src/emulator/ppu/ppu-cgb.ts

```diff
@@ -345,6 +358,38 @@ export class PpuCgb extends PpuBase {
         this.updatePalette(this.paletteOB1, value);
     };
 
+    private bgpiRead: ReadHandler = (_) => this.bgpi;
+    private bgpiWrite: WriteHandler = (_, value) => {
+        this.bgpi = value | 0x40;
+    };
+
+    private bgpdRead: ReadHandler = (_) => ((this.reg[reg.lcdc] & lcdc.enable) === 0 || this.mode !== ppuMode.draw ? this.bcram[this.bgpi & 0x3f] : 0xff);
+    private bgpdWrite: WriteHandler = (_, value) => {
+        const address = this.bgpi & 0x3f;
+
+        ((this.reg[reg.lcdc] & lcdc.enable) === 0 || this.mode !== ppuMode.draw) && (this.bcram[address] = value);
+
+        if (this.bgpi & 0x80) {
+            this.bgpi = 0x80 | (address + 1);
```

Comment:
There's a potential overflow hidden here --- you need to fix this via `((address + 1) & 0x3f)`.